core/src/test/java/io/questdb/test/griffin/GroupByTest.java

```diff
@@ -1675,6 +1675,567 @@ public void testSelectMatchingButInDifferentOrderThanGroupBy() throws Exception
+                    "1970-01-01T22:00:00.000000Z\t-0.05514933756198426\n" +
+                    "1970-01-01T22:30:00.000000Z\t-0.2093569947644236\n" +
+                    "1970-01-01T22:45:00.000000Z\t-0.8439276969435359\n" +
+                    "1970-01-01T23:00:00.000000Z\t-0.03973283003449557\n" +
+                    "1970-01-01T23:15:00.000000Z\t-0.8551850405049611\n" +
+                    "1970-01-01T23:45:00.000000Z\t0.6226001464598434\n" +
+                    "1970-01-02T00:00:00.000000Z\t-0.7195457109208119\n" +
+                    "1970-01-02T00:15:00.000000Z\t-0.23493793601747937\n" +
+                    "1970-01-02T00:30:00.000000Z\t-0.6334964081687151\n";
+            String query = "SELECT\n" +
+                    "    delivery_start_utc as y_utc_15m,\n" +
+                    "    sum(case\n" +
+                    "            when seller='sf' then -1.0*volume_mw\n" +
+                    "            when buyer='sf' then 1.0*volume_mw\n" +
+                    "            else 0.0\n" +
+                    "        end)\n" +
+                    "    as y_sf_position_mw\n" +
+                    "FROM (\n" +
+                    "    SELECT delivery_start_utc, seller, buyer, volume_mw FROM trades\n" +
+                    "    WHERE\n" +
+                    "        (seller = 'sf' OR buyer = 'sf')\n" +
+                    "    )\n" +
+                    "group by y_utc_15m " +
+                    "order by y_utc_15m";
+
+            assertQuery(expected,
+                    query,
+                    null,
+                    true,
+                    true
+            );
+
+            assertPlan(query, "Sort light\n" +
+                    "  keys: [y_utc_15m]\n" +
+                    "    GroupBy vectorized: false\n" +
+                    "      keys: [y_utc_15m]\n" +
+                    "      values: [sum(case([seller='sf',-1.0*volume_mw,buyer='sf',1.0*volume_mw,0.0]))]\n" +
+                    "        SelectedRecord\n" +
+                    "            Async JIT Filter workers: 1\n" +
+                    "              filter: (seller='sf' or buyer='sf')\n" +
+                    "                DataFrame\n" +
+                    "                    Row forward scan\n" +
+                    "                    Frame forward scan on: trades\n");
+        });
+
+
+    }
+
+    @Test
+    public void testLiftAliasesFromInnerSelect5() throws Exception {
+        // Test aliasing a function name
+        assertMemoryLeak(() -> {
+            ddl("create table x ( a int, b int, c symbol, ts timestamp ) timestamp(ts) partition by DAY WAL;");
+            insert("insert into x values (1,2,'3', now()), (2,3, '3', now()), (5,6,'4', now())");
+            insert("insert into x values (1, 5, '4', now()), (1, 3, '1', now())");
+            drainWalQueue();
+            String query =
+                    "select a, sum(b) sum, c as z, count(*) views\n" +
```

Comment:
nit: would be great to handle this case as well, so that we end up with parallel GROUP BY.